src/server/game/Anticheat/AnticheatMgr.cpp

```diff
@@ -237,22 +237,32 @@ void AnticheatMgr::SpeedHackDetection(Player* player, MovementInfo movementInfo)
     // this is the distance doable by the player in 1 sec, using the time done to move to this point.
     uint32 clientSpeedRate = distance2D * 1000 / timeDiff; // Only Chuck Norris can divide by zero so we divide by 1
 
-    // We did the (uint32) cast to accept a margin of tolerance
+    // we create a diff speed in uint32 for further precision checking to avoid legit fall and slide
+    uint32 diffspeed = clientSpeedRate - speedRate;
```

Comment:
is this handling cases where clientSpeedRate is less than speedRate, so diffspeed is a negative number stored in a uint32 ?

Reply:
> check with using + instead of -

I meant you should never use the - operator with 2 uint32 numbers where you are not sure if the result will be a positive number. uint32 is "unsigned int", "unsigned" meaning it doesn't handle negative numbers.

so, as you cannot use the - sign, you shouldn't try to get "diffspeed" at all, just do the math using + signs only.
so instead of 
```
if (diffspeed >= _assignedspeeddiff)
```
use
```
if (clientSpeedRate >= _assignedspeeddiff + speedRate)
```
which is the same logically from a math point of view, but operates in the range of uint32 (which is [0, 4 billions]

Let me show the math :P
```
uint32 diffspeed = clientSpeedRate - speedRate;
if (diffspeed >= _assignedspeeddiff)
```
```
if (clientSpeedRate - speedRate >= _assignedspeeddiff)
```
```
if (clientSpeedRate >= _assignedspeeddiff + speedRate)
```

Reply:
Math is hard :(
Haha
I will make adjustments this week. Thanks.

Reply:
btw if you don't think the speed will ever go above 2 billions ( which it should not), then you can just use `int32` instead of `uint32`

Reply:
> btw if you don't think the speed will ever go above 2 billions ( which it should not), then you can just use `int32` instead of `uint32`

at the moment we got a mix match with int and uint if i changed it to int32 for the diffspeed, issue being obvious with the clientSpeedRate needing to be uint32.